### PR TITLE
sql, toc: reorganize Compatibility and Security

### PR DESCRIPTION
**Pull request by dcalvin**
via PR:
- [sql: Organize security topics (#973)](https://github.com/pingcap/docs/pull/973)
- [sql, TOC: Reorganize Compatibility for DITA (#981)](https://github.com/pingcap/docs/pull/981)

Changes included:

- TOC
- added security-compatibility 
- divided mysql-comapatibility 
- added transaction-model

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -17,10 +17,12 @@
...
     - [TiDB 系统变量](sql/variable.md)
     - [TiDB 专用系统变量和语法](sql/tidb-specific.md)
-    - [TiDB 服务器日志文件](sql/tidb-server.md#tidb-服务器日志文件)
-    - [TiDB 访问权限管理](sql/privilege.md)
-    - [TiDB 用户账户管理](sql/user-account-management.md)
-    - [使用加密连接](sql/encrypted-connections.md)
+   + 安全   
+     - [TiDB 服务器日志文件](sql/tidb-server.md#tidb-服务器日志文件)
+     - [TiDB 访问权限管理](sql/privilege.md)
+     - [TiDB 用户账户管理](sql/user-account-management.md)
```
Comment (lilin90) [inline TOC.md]:
```suggestion
      - [TiDB 用户账户管理](sql/user-account-management.md)
```

File: TOC.md [hunk 2]
```diff
@@ -17,10 +17,12 @@
...
+     - [使用加密连接](sql/encrypted-connections.md)
```
Comment (lilin90) [inline TOC.md]:
```suggestion
      - [使用加密连接](sql/encrypted-connections.md)
```

File: TOC.md [hunk 3]
```diff
@@ -17,10 +17,12 @@
...
+     - [与 MySQL 安全特性差异](sql/security-compatibility.md)]
```
Comment (lilin90) [inline TOC.md]:
```suggestion
      - [与 MySQL 安全特性差异](sql/security-compatibility.md)]
```

File: sql/mysql-compatibility.md [hunk 4]
```diff
@@ -5,13 +5,13 @@ category: compatibility
...
+TiDB 支持 MySQL 传输协议及其绝大多数的语法。这意味着您现有的 MySQL 连接器和客户端都可以继续使用。大多数情况下您现有的应用都可以迁移至 TiDB，无需任何代码修改。
 
-包括现有的大多数 MySQL 运维工具（如 PHPMyAdmin, Navicat, MySQL Workbench 等），以及备份恢复工具（如 mysqldump, mydumper/myloader）等都可以直接使用。
+当前 TiDB 服务器官方支持的版本为 MySQL 5.7。大部分 MySQL 运维工具（如 PHPMyAdmin, Navicat, MySQL Workbench 等），以及备份恢复工具（如 mysqldump, mydumper/myloader）等都可以直接使用。
 
-不过一些特性由于在分布式环境下没法很好的实现，目前暂时不支持或者是表现与 MySQL 有差异。
+不过一些特性由于在分布式环境下没法很好的实现，目前暂时不支持或者是表现与 MySQL 有差异。一些 MySQL 语法在 TiDB 中可以解析通过，但是不会做任何后续的处理，例如 `Create Table` 语句中 `Engine` 以及 `Partition` 选项，都是解析并忽略。更多兼容性差异请参考具体的文档。
 
-一些 MySQL 语法在 TiDB 中可以解析通过，但是不会做任何后续的处理，例如 Create Table 语句中 `Engine` 以及 `Partition` 选项，都是解析并忽略。更多兼容性差异请参考具体的文档。
+> **注意：** 本页内容仅涉及 MySQL 与 TiDB 的总体差异。关于[与 MySQL 安全特性差异](../sql/security-compatibility.md)及[事务模型](../sql/transaction-model.md)的兼容信息请查看各自具体页面。
```
Comment (lilin90) [inline sql/mysql-compatibility.md]:
```suggestion
> **注意**：本页内容仅涉及 MySQL 与 TiDB 的总体差异。关于[与 MySQL 安全特性差异](../sql/security-compatibility.md)及[事务模型](../sql/transaction-model.md)的兼容信息请查看各自具体页面。
```

File: sql/mysql-compatibility.md [hunk 5]
```diff
@@ -36,15 +36,15 @@ TiDB 支持包括跨行事务、JOIN 及子查询在内的绝大多数 MySQL 5.7
 * `CREATE TABLE tblName AS SELECT stmt` 语法
 * `CREATE TEMPORARY TABLE` 语法
 * `XA` 语法（TiDB 内部使用两阶段提交，但并没有通过 SQL 接口公开）
-* `LOCK TABLE` 语法（TiDB 使用 `tidb_snapshot` 来[生成备份](../tools/mydumper.md)）
+* `LOCK TABLE` 语法（TiDB 使用 `tidb_snapshot` 来[生成备份](../tools/mydumper.md)
```
Comment (lilin90) [inline sql/mysql-compatibility.md]:
```suggestion
* `LOCK TABLE` 语法（TiDB 使用 `tidb_snapshot` 来[生成备份](../tools/mydumper.md)）
```

File: sql/mysql-compatibility.md [hunk 6]
```diff
@@ -36,15 +36,15 @@ TiDB 支持包括跨行事务、JOIN 及子查询在内的绝大多数 MySQL 5.7
...
 * `CHECK TABLE` 语法
 * `CHECKSUM TABLE` 语法
 
 ## 与 MySQL 有差异的特性
 
 ### 自增 ID
 
-TiDB 的自增 ID (Auto Increment ID) 只保证自增且唯一，并不保证连续分配。TiDB 目前采用批量分配的方式，所以如果在多台 TiDB 上同时插入数据，分配的自增 ID 会不连续。
+TiDB中，自增列只保证自增且唯一，并不保证连续分配。TiDB 目前采用批量分配 ID 的方式，所以如果在多台 TiDB 上同时插入数据，分配的自增 ID 会不连续。
```
Comment (lilin90) [inline sql/mysql-compatibility.md]:
```suggestion
TiDB 中，自增列只保证自增且唯一，并不保证连续分配。TiDB 目前采用批量分配 ID 的方式，所以如果在多台 TiDB 上同时插入数据，分配的自增 ID 会不连续。
```

File: sql/mysql-compatibility.md [hunk 7]
```diff
@@ -5,13 +5,13 @@ category: compatibility
...
 # 与 MySQL 兼容性对比
 
-TiDB 支持包括跨行事务、JOIN 及子查询在内的绝大多数 MySQL 5.7 的语法，用户可以直接使用现有的 MySQL 客户端连接。如果现有的业务已经基于 MySQL 开发，大多数情况不需要修改代码即可直接替换单机的 MySQL。
+TiDB 支持 MySQL 传输协议及其绝大多数的语法。这意味着您现有的 MySQL 连接器和客户端都可以继续使用。大多数情况下您现有的应用都可以迁移至 TiDB，无需任何代码修改。
 
-包括现有的大多数 MySQL 运维工具（如 PHPMyAdmin, Navicat, MySQL Workbench 等），以及备份恢复工具（如 mysqldump, mydumper/myloader）等都可以直接使用。
+当前 TiDB 服务器官方支持的版本为 MySQL 5.7。大部分 MySQL 运维工具（如 PHPMyAdmin, Navicat, MySQL Workbench 等），以及备份恢复工具（如 mysqldump, mydumper/myloader）等都可以直接使用。
 
-不过一些特性由于在分布式环境下没法很好的实现，目前暂时不支持或者是表现与 MySQL 有差异。
+不过一些特性由于在分布式环境下没法很好的实现，目前暂时不支持或者是表现与 MySQL 有差异。一些 MySQL 语法在 TiDB 中可以解析通过，但是不会做任何后续的处理，例如 `Create Table` 语句中 `Engine` 以及 `Partition` 选项，都是解析并忽略。更多兼容性差异请参考具体的文档。
```
Comment (lilin90) [inline sql/mysql-compatibility.md]: Or we provide the specific doc?

```suggestion
不过一些特性由于在分布式环境下没法很好的实现，目前暂时不支持或者是表现与 MySQL 有差异。一些 MySQL 语法在 TiDB 中可以解析通过，但是不会做任何后续的处理，例如 `Create Table` 语句中 `Engine` 以及 `Partition` 选项，都是解析并忽略。
```

File: sql/mysql-compatibility.md [hunk 8]
```diff
@@ -100,61 +100,14 @@ TiDB 实现了 F1 的异步 Schema 变更算法，DDL 执行过程中不会阻
...
     - 在修改类型定义方面，支持的包括 default value，comment，null，not null 和 OnUpdate。
-    - 支持 LOCK [=] {DEFAULT|NONE|SHARED|EXCLUSIVE} 语法，但是不做任何事情（pass through）。
+    - 支持 LOCK [=] {DEFAULT|NONE|SHARED|EXCLUSIVE} 语法，但是不做任何操作（pass through）。
     - 不支持对enum类型的列进行修改
 
-### 事务模型
+### 数据库管理
 
-TiDB 使用乐观事务模型，在执行 `Update`、`Insert`、`Delete` 等语句时，只有在提交过程中才会检查写写冲突，而不是像 MySQL 一样使用行锁来避免写写冲突。类似的，诸如 `GET_LOCK()` 和 `RELEASE_LOCK()` 等函数以及 `SELECT .. FOR UPDATE` 之类的语句在 TiDB 和 MySQL 中的执行方式并不相同。所以业务端在执行 SQL 语句后，需要注意检查 commit 的返回值，即使执行时没有出错，commit 的时候也可能会出错。
++ [`ANALYZE TABLE`](/sql/statistics.md#manual-collection) 语句在 TiDB 和 MySQL 中表现不同。在 MySQL/InnoDB 中，它是一个轻量级语句，执行过程较短；而在 TiDB 中，它会完全重构表的统计数据，语句执行过程较长。
```
Comment (lilin90) [inline sql/mysql-compatibility.md]:
```suggestion
+ [`ANALYZE TABLE`](/sql/statistics.md#手动收集) 语句在 TiDB 和 MySQL 中表现不同。在 MySQL/InnoDB 中，它是一个轻量级语句，执行过程较短；而在 TiDB 中，它会完全重构表的统计数据，语句执行过程较长。
```

File: sql/security-compatibility.md  [hunk 9]
```diff
@@ -0,0 +1,17 @@
+---
+title: 与 MySQL 安全特性差异
+category: user guide
+---
+
+# 与 MySQL 安全特性差异
+
+除以下功能外，TiDB 支持与 MySQL 5.7 类似的安全特性。
+
+- 仅支持 `mysql_native_password` 鉴权方案。
```
Comment (lilin90) [inline sql/security-compatibility.md ]: The word "鉴权" is seldom used in our docs. "身份验证" is a choice?

File: sql/security-compatibility.md  [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+- 不支持外部鉴权方式（如 LDAP）。
+- 不支持列级别权限设置。
+- 不支持使用证书鉴权。 [#9708](https://github.com/pingcap/tidb/issues/9708)
```
Comment (lilin90) [inline sql/security-compatibility.md ]:
```suggestion
- 不支持使用证书鉴权。[#9708](https://github.com/pingcap/tidb/issues/9708)
```

File: sql/security-compatibility.md  [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+- 不支持密码过期，最后一次密码变更记录以及密码生存期。 [#9709](https://github.com/pingcap/tidb/issues/9709)
```
Comment (lilin90) [inline sql/security-compatibility.md ]:
```suggestion
- 不支持密码过期，最后一次密码变更记录以及密码生存期。[#9709](https://github.com/pingcap/tidb/issues/9709)
```

File: sql/security-compatibility.md  [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+- 不支持权限属性 `max_questions`，`max_updated`，`max_connections` 以及 `max_user_connections`。
+- 不支持密码验证。 [#9741](https://github.com/pingcap/tidb/issues/9741)
```
Comment (lilin90) [inline sql/security-compatibility.md ]:
```suggestion
- 不支持密码验证。[#9741](https://github.com/pingcap/tidb/issues/9741)
```

File: sql/security-compatibility.md  [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+- 不支持透明数据加密（TDE）。
```
Comment (lilin90) [inline sql/security-compatibility.md ]:
```suggestion
- 不支持透明数据加密 (TDE)。
```

File: sql/security-compatibility.md  (hunk 9 above)
Comment (dcalvin) [reply sql/security-compatibility.md ]: Sounds good

**Comment by dcalvin** [comment]
Thanks @lilin90. PTAL again.

File: TOC.md [hunk 14]
```diff
@@ -18,9 +18,11 @@
...
     - [TiDB 专用系统变量和语法](sql/tidb-specific.md)
     - [TiDB 服务器日志文件](sql/tidb-server.md#tidb-服务器日志文件)
-    - [TiDB 访问权限管理](sql/privilege.md)
-    - [TiDB 用户账户管理](sql/user-account-management.md)
-    - [使用加密连接](sql/encrypted-connections.md)
+    + 安全
+      - [TiDB 访问权限管理](sql/privilege.md)
+      - [TiDB 用户账户管理](sql/user-account-management.md)
+      - [使用加密连接](sql/encrypted-connections.md)
+      - [与 MySQL 安全特性差异](sql/security-compatibility.md)]
```
Comment (lilin90) [inline TOC.md]:
```suggestion
      - [与 MySQL 安全特性差异](sql/security-compatibility.md)
```

File: sql/security-compatibility.md  (hunk 9 above)
Comment (lilin90) [inline sql/security-compatibility.md ]:
```suggestion
- 仅支持 `mysql_native_password` 身份验证方案。
```

**Pull request merged**